addons/portal/static/src/js/portal_name_and_signature.js

```diff
@@ -0,0 +1,582 @@
+        this.$signatureField.css({width: 'unset'});
+        var width = this.$signatureField.width();
+        var height = parseInt(width / this.signatureRatio);
+
+        // necessary because the lib is adding invisible div with margin
+        // signature field too tall without this code
+        this.$signatureField.css({
+            width: width,
+            height: height,
+        });
+
+        this.$signatureField
+            .empty()
+            .jSignature({
+                'decor-color': '#D1D0CE',
+                'background-color': '#FFF',
+                'color': '#000',
+                'lineWidth': 2,
+                'width': width,
+                'height': height,
+            });
+
+        this.emptySignature = this.$signatureField.jSignature('getData');
+
+        this.setMode(this.signMode, true);
+
+        this.focusName();
+
+        return $.Deferred().resolve();
+    },
+    /**
+     * Changes the signature mode. Toggles the display of the relevant
+     * controls and resets the drawing.
+     *
+     * @param {string} mode - the mode to use. Can be one of the following:
+     *  - 'draw': the user draws the signature manually with the mouse
+     *  - 'auto': the signature is drawn automatically using a selected font
+     *  - 'load': the signature is loaded from an image file
+     * @param {boolean} [reset=false] - Set to true to reset the elements
+     *  even if the @see mode has not changed. By default nothing happens
+     *  if the @see mode is already selected.
+     */
+    setMode: function (mode, reset) {
+        if (reset !== true && mode === this.signMode) {
+            // prevent flickering and unnecessary compute
+            return;
+        }
+
+        this.signMode = mode;
+
+        this.$drawClear.toggleClass('d-none', this.signMode !== 'draw');
+        this.$autoSelectStyle.toggleClass('d-none', this.signMode !== 'auto');
+        this.$loadFile.toggleClass('d-none', this.signMode !== 'load');
+
+        this.$drawButton.toggleClass('active', this.signMode === 'draw');
+        this.$autoButton.toggleClass('active', this.signMode === 'auto');
+        this.$loadButton.toggleClass('active', this.signMode === 'load');
+
+        this.$signatureField.jSignature(this.signMode === 'draw' ? 'enable' : 'disable');
+        this.$signatureField.jSignature('reset');
+
+        if (this.signMode === 'auto') {
+            // draw based on name
+            this._drawCurrentName();
+        } else {
+            // close style dialog
+            this.$autoFontSelection.addClass('d-none');
+        }
+
+        if (this.signMode !== 'load') {
+            // close invalid file alert
+            this.$loadInvalid.addClass('d-none');
+        }
+    },
+    /**
+     * Gets the current name and signature, validates them, and returns
+     * the result. If they are invalid, displays the errors to the user.
+     *
+     * @return {boolean} whether the current name and signature are valid
```

Comment:
`@returns`

Reply:
ok (we should really automate the detection of those mistakes)

Reply:
Well, they are not really mistakes except in my mind :wink: But maybe one day we will compile the JS Doc... :)